### PR TITLE
SJSU-1039-Sorting/Filtering

### PR DESCRIPTION
**Pull request by tienquocnguyen**
Implemented the sorting/filtering, please leave me any suggestions.

### CODE REVIEW

**Comment by tienquocnguyen** [comment]
tested with
destionationName
lastIndex
pageNumber
numResults
numberRooms
checkIn
checkOut
free_wifi
pool
free_breakfast
free_parking
pet_friendly
star_rating
review_score
price_low
price_high
sortObject

File: routes/api/hotels.js [hunk 1]
```diff
@@ -8,23 +8,101 @@ const Hotel = require("../../models/Hotel");
 const Booking = require("../../models/booking");
```
Comment (DaleCS) [inline routes/api/hotels.js]: This import isn't being used.

File: routes/api/hotels.js [hunk 2]
```diff
@@ -8,23 +8,101 @@ const Hotel = require("../../models/Hotel");
...
         checkin:req.query.checkIn.replace('"','').replace('"',''),
         checkout:req.query.checkOut.replace('"','').replace('"',''),
     };
     var numberRooms = parseInt(req.query.numberRooms);
     var startIndex = req.query.lastIndex;
-
     const NUM_RESULTS = req.query.numResults;
-
     const regex = new RegExp(searchKey,"ig");
     //.split("").join('*')
```
Comment (DaleCS) [inline routes/api/hotels.js]: Unused comments.

File: routes/api/hotels.js [hunk 3]
```diff
@@ -8,23 +8,101 @@ const Hotel = require("../../models/Hotel");
...
+        amenities: free_parking,
+        amenities: pet_friendly,
+        amenities: free_breakfast,
+        $and:[{'price.singlePrice': {$gt: price_low}}, {'price.singlePrice': {$lt: price_high}}],
+        star_rating: {$gt: star_rating},
+        guest_rating: {$gt: review_score},
+        $or:[{name:regex}, {city:regex},{airports:regex}]
+    }).sort(sortByObject).then((doc,err)=>{
         if(err) res.status(400).json(err);
         // var startIndex = 5 * pageNumber - 5;
```
Comment (DaleCS) [inline routes/api/hotels.js]: Unused comment

File: routes/api/hotels.js [hunk 4]
```diff
@@ -1,40 +1,109 @@
...
-        // var startIndex = 5 * pageNumber - 5;
         var result = [];
         let bookingID = "bookid"
         while(result.length < NUM_RESULTS && doc[startIndex] !== undefined){
             var arr = doc[startIndex]
-            let singleAvaliable = checkAvalibity(doc[startIndex].roomTypeAndNumber.single, date, numberRooms, bookingID);
-            let doubleAvaliable = checkAvalibity(doc[startIndex].roomTypeAndNumber.double, date, numberRooms, bookingID);
-            let kingAvaliable = checkAvalibity(doc[startIndex].roomTypeAndNumber.king, date, numberRooms, bookingID);
-            let studioAvaliable = checkAvalibity(doc[startIndex].roomTypeAndNumber.studio, date, numberRooms, bookingID);
+            let singleAvaliable = checkAvailable(doc[startIndex].roomTypeAndNumber.single, date, numberRooms, bookingID);
```
Comment (DaleCS) [inline routes/api/hotels.js]: Misspelled word: 'Avaliable'

**Comment by tienquocnguyen** [comment]
> Can you explain how a tester can test this in their machine with Postman?

Do nodemon start in the terminal, then go to postman, change from GET to POST. Enter localhost:5000/api/hotel/search. Enter the parameters above with the values you want. Date needs to be in yyyy-mm-dd format. Then, press send and it should show you the hotels. If you want to look at the specific amenities then you would need to go inside the code, in 'item' add amenities: arr.amenities. Sample values are attached.
Screen Shot 2019-03-30 at 12 15 56 PM

**Pull request merged**